src/main/java/org/alfresco/rest/api/impl/NodesImpl.java

```diff
@@ -1319,6 +1322,7 @@ public Node get(int index)
                     calculateRelativePath(parentFolderNodeId, node);
                 }
 
+                node.setIsFavorite(isFavorite(node));
```

Comment:
Should be optionally included ??

Reply:
we investigate the performance problem and simplify the calculation call of isFavorite.  The time in bamboo for tests is not modified. But if are too many fields and we need only minimal information I can change to use include.

Reply:
This should be optionally included. The current implementation means each isFavorite call extracts all favourite nodes from the preferences service and checks for their existence. I would imagine this can have a performance hit as each node returned invokes isFavorite. Ideally we'd hit preferences service once, not check for existence on favourite noderefs (instead, rely on calling get node, children for existence/permission check).

Reply:
Also wonder if the setting of isFavorite on a node can be moved to the getFolderOrDocumentFullInfo method. This seems to be the place where all other node construction take place, and also where the include parameters are provided.